front/app/containers/Admin/projects/project/nativeSurvey/FormResults/FormResultsQuestion/TextQuestion/Analysis/index.tsx

```diff
@@ -47,9 +61,76 @@ const Analysis = ({ customFieldId }: { customFieldId: string }) => {
     addAnalysis,
   ]);
 
+  const toggleDropdown = () => {
+    setDropdownOpened(!dropdownOpened);
+  };
+
   return (
     <Box>
-      {relevantAnalysis && <AnalysisInsights analysis={relevantAnalysis} />}
+      {relevantAnalysis && relevantAnalysis.attributes.show_insights && (
```

Comment:
We are fetching the list of analyses for the phase and then filter them to find the relevant one for the specific question. We could implement a back-end filter for this indeed but for now I don't think it's needed because it's easy to filter this on the front-end and it means we only make one request. Otherwise, we need to make a separate api call for each question which would not be ideal.

Reply:
In the solution I had in mind it would be one API call for all questions together (unless I'm overseeing something). But sure, this is fine for me :)